plugins/plugin-webpack/plugin.js

```diff
@@ -102,7 +108,10 @@ function emitHTMLFiles({doms, jsEntries, stats, baseUrl, buildDirectory, htmlMin
       ? minify(dom.serialize(), htmlMinifierOptions)
       : dom.serialize();
 
-    fs.writeFileSync(path.join(buildDirectory, htmlFile), html);
+    const outputFile = path.join(outputDirectory, htmlFile);
+    // If the user specified a different output, we may not have an existing folder structure
+    ensureDirectoryExists(path.dirname(outputFile))
+    fs.writeFileSync(outputFile, html);
```

Comment:
What happens if the `output.path` is not specified on webpack config? @jalovatt

Reply:
The plugin sets a default output path (line 402 in this PR):
```
      const extendedConfig = extendConfig({
        ...webpackConfig,
        plugins,
        entry,
        output: {
          path: buildDirectory,
          publicPath: baseUrl,
          filename: jsOutputPattern,
        },
      });
 ``
`extendConfig` (line 271 in this PR) accepts either a function or an object. With a function, the output path will always be set as long as the user is only overriding that property of `output`:
```
(config) => { config.output.path = path.resolve(__dirname, 'dist'); }
```
If the user replaces `output` though, or passes a config object, `extendConfig` just uses a simple spread to merge it in:
```
(cfg) => ({ ...cfg, ...args.extendConfig ])
```
In this case, if the user changes anything in `output` they would have to provide all of `{ path, publicPath, filename }` for the plugin to work correctly. (Not a new behaviour in this PR)